modules/subgraphs/gauge-subgraph/gauge-subgraph.service.ts

```diff
@@ -238,6 +244,29 @@ export class GaugeSubgraphService {
         return meta;
     }
 
+    public async getGaugesInfo(poolIds: string[]): Promise<LiquidityGaugesInfo> {
+        const { liquidityGauges } = await this.sdk.GaugesInfo({ poolIds });
+        return liquidityGauges;
+    }
+
+    public async getGaugesStreamers(poolIds: string[]) {
+        const { liquidityGauges } = await this.sdk.GaugeStreamers({ poolIds });
+        return liquidityGauges;
+    }
+
+    public async getRootGaugesInfo(recipients: string[], chain: Chain): Promise<RootGaugesInfo> {
+        const currentChainId = getRequestScopeContextValue('chainId');
+        // Run query in MAINNET subgraph
+        // Can we do it in a cleaner way?
+        setRequestScopedContextValue('chainId', '1');
```

Comment:
As far as I understand, the current architecture assumes that each job runs in a concrete network so it will setup the subgraph url depending on the current context network. However in this case (RootGauges), we need to query the MAINNET gauges subgraph independently of the current context network. Is there a better solution?

Reply:
I had to do something similar for the veBAL sync. I created a subgraph service called `vebal-locks-subgraph` which basically just points to the mainnet gauges subgraph. I'd suggest you rename that to `mainnet-gauges-subgraph` and also use it to get the root gauges.

Reply:
But I would actually argue not to use the subgraph but rather the gauge controller contract https://etherscan.io/address/0xC128468b7Ce63eA702C1f104D55A2566b13D3ABD#readContract